### ISSUE TITLE
Upgrade Needed on Social Media API Version

### ISSUE THREAD

**Issue by KenRHill** [YouPHPTube/YouPHPTube#1736]
Hi Daniel,

I have a strange issue with the social media login options on YouPHPTube.  I am able to successfully login to my streamer site from a computer and web browser using Facebook, Twitter, and LinkedIn buttons on the steamer site.  

When I login using the new mobile app however, if I am using Facebook, Twitter or LinkedIn from the Android mobile app, it does not allow the login to complete.

I went over to the developer accounts at Facebook, Twitter and LinkedIn to check to make sure that all of my URI and OAuth settings are correct and that both the website and the mobile app URIs are added.  The URI's all match up to what is shown by the YouPHPTube plugins respectively, however I am still unable to login from the mobile app using Facebook, Twitter or LinkedIn as each social media login throws an error.

It apperars that Facebook, Twitter, LinkedIn etc. may now be running upgraded API versions.  They now appear to be on API v3.1, v3.2 or v3.3.  It is my understanding that YouPHPTube is using API v2.  

Am I correct in this assumption? If so, is it possible to upgrade the YouPHPTube API version to comply with the v3.1, v3.2 and v3.3 standards or is there something else I need to do to fix my Social Media logins using the mobile app?  Please advise.  

Screenshot of Facebook Error
![Facebook](https://user-images.githubusercontent.com/37643777/57975539-cab24900-797f-11e9-9a0a-847b59fb861f.jpg)

Screenshot of Twitter Error
![Twitter](https://user-images.githubusercontent.com/37643777/57975542-d6057480-797f-11e9-8458-56f0d63ff175.jpg)

Screenshot of LinkedIn Error
![LinkedIn](https://user-images.githubusercontent.com/37643777/57975545-dc93ec00-797f-11e9-9fcd-883767635f77.jpg)

**Comment by Puyodead1** [comment]
My guess is that its a different redirect url on mobile then on a browser

**Comment by DanielnetoDotCom** [comment]
I guess @Puyodead1 is correct

Facebook, make sure you also add the mobilemanager URL

Linkedin: there is an update on Linkedin (this month) we still not able to figure out yet but we are aware and work on it.

also make sure you add the mobilemanager URL in all of the OAuth authentications.

![image](https://user-images.githubusercontent.com/18009293/58020258-03683480-7ade-11e9-8f24-1ffcb9384d7c.png)

**Comment by KenRHill** [comment]
Facebook just put some kind of restriction on my app stating that it was causing a bad experience on their site.  I have checked to make sure that the two URIs for YouPHPTube are added to the Facebook Developer app settings section as shown.

![ValidOAuth](https://user-images.githubusercontent.com/37643777/58123517-d3de2880-7bc0-11e9-86cd-2f15f9d62808.JPG)

**Comment by DanielnetoDotCom** [comment]
I am not sure what is going on, I have never seen this before.

maybe the video content that you have.

Did you try to appeal?

**Comment by KenRHill** [comment]
After some tinkering in all three social media developer accounts, I believe I just figured out the issue for all three, although I am not 100% sure why this is now required when it used to work before.  Maybe updates to their systems?  I have no idea, but what fixed it for me is as follows:

As you can see from the previous posts above... YouPHPTube plugin suggests to use the OAuth URI paths https://www.videorefuge.com/objects/login.json.php?type=Facebook and https://www.videorefuge.com/plugin/MobileManager/oauth2.php?type=Facebook.  In order to make it work, I had to also specify https://videorefuge.com/objects/login.json.php?type=Facebook and https://videorefuge.com/plugin/MobileManager/oauth2.php?type=Facebook

In other words I logged into the developer console of all three social media networks and I added the URIs by taking out the www. from the URI path.  It is now working for logins.

**Comment by DanielnetoDotCom** [comment]
yes, you right, you can add both authorizations: with www and without it
